docs/pages/guides/replicating-onchain-state.mdx

```diff
@@ -0,0 +1,126 @@
+# Replicating onchain state from MUD Store events
+
+This guide captures how you might recreate onchain state from MUD’s Store events. We use this pattern in our sync stack to hydrate a client or indexer database from blockchain logs fetched from an RPC.
+
+## Store: MUD’s onchain storage
+
+First, it’s helpful to understand how MUD stores its data on chain. Like a native Solidity `mapping`, MUD stores values at keys. However, it handles both keys and values in a particular way to 1) give a bit more flexibility in data modeling and 2) squeeze out a bit more gas.
+
+- In MUD, keys are represented as a combination of table ID and key tuple (i.e. a composite key). Internally, the storage slot is derived from a hash of the table ID, key tuple, and “section” of data (static or dynamic).
+- All static (fixed length) fields are tightly packed into a single bytes value stored in a single storage slot. Two `uint32` fields and a `bool` field would be packed into 9 bytes (4 + 4 + 1).
+- For all dynamic (variable length) fields, we encode and store a “dynamic data layout” (also called “encoded lengths”) that represents the length of each dynamic field (i.e. number of items in an array, length of string or bytes). Each dynamic field is given its own storage slot that contains its value.
```

Comment:
These bullet points are great (very glad to have an explanation of this stuff), but it's a lot to process on first glance with all the new terms, brackets, and quotation marks. 

A diagram or even glossary of terms could help!

Reply:
I've reworded this and now link out to the other docs pages that provide more detail on the encoding.

Reply:
Super clear and to the point now!! Sad you had to remove your text, maybe it can fit elsewhere